pandas/core/indexes/datetimelike.py

```diff
@@ -236,12 +236,44 @@ def freq(self):
 
     @freq.setter
     def freq(self, value):
+        msg = ('Setting {obj}.freq has been deprecated and will be removed '
+               'in a future version; use {obj}.set_freq instead.'
+               ).format(obj=type(self).__name__)
+        warnings.warn(msg, FutureWarning, stacklevel=2)
         if value is not None:
             value = frequencies.to_offset(value)
             self._validate_frequency(self, value)
 
         self._freq = value
 
+    def set_freq(self, freq):
```

Comment:
can you make this available on the .dt accessors as well

Reply:
This currently does nothing when used with the `dt` accessor due to #20937; the underlying frequency gets changed with `set_freq`, but upon `Series` construction the new frequency gets lost.

I've pushed a commit to add this to the `dt` accessor, and xfailed the corresponding tests I've added.  Can revert the commit if this is no longer deemed important.